cabal-install/Distribution/Solver/Modular/Log.hs

```diff
@@ -72,17 +77,21 @@ logToProgress mbj l = let
     -- exhaustiveness and first conflict set.
     go :: Progress Message a b
        -> Progress Message a b
-       -> Progress String (Bool, Maybe (ConflictSet QPN)) b
+       -> Progress String (Exhaustive, ConflictSet QPN, ConflictMap) b
        -> Progress String String b
     go ms (Step _ ns) (Step x xs)           = Step x (go ms ns xs)
     go ms r           (Step x xs)           = Step x (go ms r  xs)
-    go ms _           (Fail (exh, Just cs)) = Fail $
-                                              "Could not resolve dependencies:\n" ++
-                                              unlines (messages $ showMessages (L.foldr (\ v _ -> v `CS.member` cs) True) False ms) ++
-                                              (if exh then "Dependency tree exhaustively searched.\n"
-                                                      else "Backjump limit reached (" ++ currlimit mbj ++
-                                                               "change with --max-backjumps or try to run with --reorder-goals).\n")
-                                                  where currlimit (Just n) = "currently " ++ show n ++ ", "
-                                                        currlimit Nothing  = ""
+    go ms _           (Fail (exh, cs, cm))  = Fail $
+        "Could not resolve dependencies:\n" ++
+        unlines (messages $ showMessages (L.foldr (\ v _ -> v `CS.member` cs) True) False ms) ++
```

Comment:
Okay, so to summarize:
* We always print the first error encountered.
* In case the backjump limit is reached, we print the first conflict set (which coincides with the first error).
* For an exhaustive search, we print the final conflict set (which may not coincide with the first error).

So this could be solved
1. either by choosing to print the first rather than the final conflict set also in the exhaustive case (what would this imply?),
2. or by printing the error only in the non-exhaustive case and thus avoiding the problem (in fact, in an early version of the PR, @kosmikus and I had printed the error message only in the non-exhaustive case, but we changed it),
3. or by printing the last error in the non-exhaustive case.

My understanding of the issue is still somewhat incomplete, but it seems to me that using the reasoning from #941 also applies to the exhaustive case, so option (1) looks to me like the soundest one?